lib/matplotlib/backends/backend_svg.py

```diff
@@ -298,6 +298,24 @@ def generate_css(attrib=None):
 _capstyle_d = {'projecting': 'square', 'butt': 'butt', 'round': 'round'}
 
 
+def _check_is_str(info, key):
+    if not isinstance(info, str):
+        raise TypeError(f'Invalid type for {key} metadata. Expected str, not '
+                        f'{type(info)}.')
+
+
+def _check_is_iterable_of_str(infos, key):
+    for info in infos:
+        if not isinstance(info, str):
+            raise TypeError(f'Invalid type for {key} metadata. Expected '
+                            f'iterable of str, not {type(info)}.')
+
+
+def _raise_not_str_or_iterable_of_str(infos, key):
+    raise TypeError(f'Invalid type for {key} metadata. Expected str or '
+                    f'iterable of str, not {type(infos)}.')
```

Comment:
These two could be joined, I think? One or the other is called based on `np..iterable`, which could be in a single function here.

Reply:
You are correct. Well spotted!